### PR TITLE
[SDK] Implemented getEscrow and getDetails method updated

### PR DESCRIPTION
**Pull request by eugenvoronov**
## Description
Implemented the ability to retrieve a single escrow's data by escrow address. Now associated escrow data can be fetched without relying on other filters.

## Summary of changes
**Added getEscrow function in escrowClient.**
- This function allows fetching an escrow by escrow address.
- It uses the GET_ESCROW_BY_ADDRESS_QUERY GraphQL query to get the data.
- An error is thrown if an invalid ethereum address is provided.
- Returns null if the escrow isn't found.

**Added unit tests for getEscrowByAddress.**
- Tests cover scenarios where an invalid address is provided, a valid address with no corresponding escrow, and a valid address with a corresponding escrow.

**Modified GraphQL queries to include GET_ESCROW_BY_ADDRESS_QUERY.**

**Updated getDetails method in Job Launcher Server**

**Changed NestExpressApplication -> INestApplication**
`NestExpressApplication` does not include express-specific methods and properties.

## How test the changes
`yarn test`

## Related issues
[[Job Launcher] [SDK] Implement getEscrow method to SDK and use it in getDetails method#892](https://github.com/humanprotocol/human-protocol/issues/892)

### CODE REVIEW

**Review by flopez7 — APPROVED** [review]

File: packages/apps/job-launcher/server/src/modules/job/job.dto.ts [hunk 1]
```diff
@@ -160,15 +160,15 @@ export class ManifestDetails {
 
   @IsNotEmpty()
   @IsString()
-  exchangeOracleAddress: string;
+  exchangeOracleAddress?: string;
```
Comment (flopez7) [inline packages/apps/job-launcher/server/src/modules/job/job.dto.ts:161]: The var is nullable, but the validator forces to have a value

File: packages/apps/job-launcher/server/src/modules/job/job.dto.ts [hunk 2]
```diff
@@ -160,15 +160,15 @@ export class ManifestDetails {
...
 
   @IsNotEmpty()
   @IsString()
-  recordingOracleAddress: string;
+  recordingOracleAddress?: string;
```
Comment (flopez7) [inline packages/apps/job-launcher/server/src/modules/job/job.dto.ts:165]: Same here

File: packages/apps/job-launcher/server/src/modules/job/job.dto.ts [hunk 3]
```diff
@@ -160,15 +160,15 @@ export class ManifestDetails {
...
 
   @IsNotEmpty()
   @IsString()
-  reputationOracleAddress: string;
+  reputationOracleAddress?: string;
```
Comment (flopez7) [inline packages/apps/job-launcher/server/src/modules/job/job.dto.ts:169]: And here

**Review by leric7 — CHANGES_REQUESTED** [review]
We are moving `getEscrows` function to `EscrowUtils` of the SDK. I think it'll be better define this new function in that module as well. We'll need `chainId` as function argument.

**Comment by portuu3** [comment]
@eugenvoronov please create a task for getEscrow function on python SDK and also sdk docs have to be updated

**Comment by eugenvoronov** [comment]
[@flopez7 ](https://github.com/humanprotocol/human-protocol/issues/915)

**Review by flopez7 — CHANGES_REQUESTED** [review]

**Review by flopez7 — DISMISSED** [review]

**Review by leric7 — DISMISSED** [review]

**Pull request merged**